skimage/morphology/misc.py

```diff
@@ -235,3 +244,212 @@ def remove_small_holes(ar, area_threshold=64, connectivity=1, *, out=None):
+    2. Sort indices by their label value, ensuring that indices which point to
+       the same object are next to each other. This optimization allows finding
+       all parts of an object, simply by stepping to the neighboring indices.
+    3. Sort boundary indices by `priority`. Use a stable-sort to preserve the
+       ordering from the previous sorting step. If `priority` is not given,
+       use :func:`numpy.bincount` as a fallback.
+    4. Construct a :class:`scipy.spatial.cKDTree` from the boundary indices.
+    5. Iterate across boundary indices in priority-sorted order, and query the
+       kd-tree for objects that are too close. Remove ones that are and don't
+       take them into account when evaluating other objects later on.
+
+    The performance of this algorithm depends on the number of samples in
+    `label_image` that belong to an object's border.
+
+    Examples
+    --------
+    >>> import skimage as ski
+    >>> ski.morphology.remove_objects_by_distance(np.array([2, 0, 1, 1]), 2)
+    array([0, 0, 1, 1])
+    >>> ski.morphology.remove_objects_by_distance(
+    ...     np.array([2, 0, 1, 1]), 2, priority=np.array([0, 1, 9])
+    ... )
+    array([2, 0, 0, 0])
+    >>> label_image = np.array(
+    ...     [[8, 0, 0, 0, 0, 0, 0, 0, 0, 9, 9],
+    ...      [8, 8, 8, 0, 0, 0, 0, 0, 0, 9, 9],
+    ...      [0, 0, 0, 0, 0, 0, 0, 0, 9, 0, 0],
+    ...      [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+    ...      [0, 0, 3, 0, 0, 0, 1, 0, 0, 0, 0],
+    ...      [2, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0],
+    ...      [0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0],
+    ...      [0, 0, 0, 0, 0, 0, 0, 0, 0, 7, 7]]
+    ... )
+    >>> ski.morphology.remove_objects_by_distance(
+    ...     label_image, min_distance=3
+    ... )
+    array([[8, 0, 0, 0, 0, 0, 0, 0, 0, 9, 9],
+           [8, 8, 8, 0, 0, 0, 0, 0, 0, 9, 9],
+           [0, 0, 0, 0, 0, 0, 0, 0, 9, 0, 0],
+           [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+           [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+           [2, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+           [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+           [0, 0, 0, 0, 0, 0, 0, 0, 0, 7, 7]])
+    """
+    if min_distance < 0:
+        raise ValueError(f"min_distance must be >= 0, was {min_distance}")
+    if not np.issubdtype(label_image.dtype, np.integer):
+        raise ValueError(
+            f"`label_image` must be of integer dtype, got {label_image.dtype}"
+        )
+    if out is None:
+        out = label_image.copy(order="C")
+    elif out is not label_image:
+        out[:] = label_image
+    # May create a copy if order is not C, account for that later
+    out_raveled = out.ravel(order="C")
+
+    if spacing is not None:
+        spacing = np.array(spacing)
+        if spacing.shape != (out.ndim,) or spacing.min() <= 0:
+            raise ValueError(
+                "`spacing` must contain exactly one positive factor "
+                "for each dimension of `label_image`"
+            )
+
+    indices = np.nonzero(out_raveled)[0]
```

Comment:
```suggestion
    indices = np.flatnonzero(out_raveled)
```

Unless this is deprecated in np2, but I don't think so...?

Reply:
Oh, nice. I think it's not.